pip/pip-437.md

```diff
@@ -0,0 +1,129 @@
+# PIP-437: Granular and Fixed-Delay Policies for Message Delivery
+
+# Background
+Pulsar's delayed delivery feature allows producers to schedule messages to be delivered at a future time. To provide administrative control and prevent abuse, **PIP-315** introduced a `maxDeliveryDelayInMillis` policy.
+While this provides an important safeguard by setting an *upper bound*, it does not address all administrative needs. There is currently no way to enforce a *specific, non-overridable* delay for all messages on a topic.
+
+# Motivation
+
+### The Limitation:
+Pulsar's delayed message delivery relies on tracking individual messages until their delivery time. This tracking state is persisted as part of the subscription's cursor metadata,
+which records gaps or "holes" for unacknowledged messages. The Pulsar Managed Ledger has a hard limit on the number of disjoint unacknowledged ranges it can persist for a cursor, 
+configured by `managedLedgerMaxUnackedRangesToPersist` (defaulting to 10,000).
+A large volume of messages with widely varying delivery times can easily exhaust this capacity with each delayed message creating a separate unacknowledged hole. 
```

Comment:
It would also be useful to document the benefits of the fixed delivery delay. There's a comment https://github.com/apache/pulsar/pull/24625#issuecomment-3232734535 with some content that would be useful to include in the PIP document.